common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -143,7 +143,11 @@ public static boolean execICU(final UTF8String l, final UTF8String r,
    * Collation-aware regexp expressions.
    */
 
-  // TODO: Add more collation-aware regexp expressions.
+  private static final UTF8String lowercaseRegexPrefix = UTF8String.fromString("(?i)");
```

Comment:
We need both "u" and "i" flags, per [documentation](https://docs.oracle.com/javase/tutorial/essential/regex/pattern.html) for "u" flag:
> When this flag is specified then case-insensitive matching, when enabled by the CASE_INSENSITIVE flag, is done in a manner consistent with the Unicode Standard. By default, case-insensitive matching assumes that only characters in the US-ASCII charset are being matched.